plugins/orchestrator-backend/src/service/SonataFlowService.ts

```diff
@@ -104,15 +106,21 @@ export class SonataFlowService {
         ? `${args.serviceUrl}/${args.definitionId}?businessKey=${args.businessKey}`
         : `${args.serviceUrl}/${args.definitionId}`;
 
-      const result = await fetch(urlToFetch, {
+      const response = await fetch(urlToFetch, {
         method: 'POST',
         body: JSON.stringify(args.inputData),
         headers: { 'content-type': 'application/json' },
       });
 
-      const json = await result.json();
-      this.logger.debug(`Execute workflow result: ${JSON.stringify(json)}`);
-      return json;
+      if (response.ok) {
+        const json = await response.json();
+        this.logger.debug(`Execute workflow result: ${JSON.stringify(json)}`);
+        return json;
+      }
+      const res = await response.json();
+      throw new Error(
```

Comment:
by chance it reproduced for me today during development with this response:

Response {
backend:start:   status: 400,
backend:start:   statusText: 'Bad Request',
backend:start:   headers: Headers { 'content-type': 'application/json', 'content-length': '271' },
backend:start:   body: ReadableStream { locked: false, state: 'readable', supportsBYOB: true },
backend:start:   bodyUsed: false,
backend:start:   ok: false,
backend:start:   redirected: false,
backend:start:   type: 'basic',
backend:start:   url: 'http://localhost:8899/spring-boot-backend'
backend:start: }

so sometimes the data is in statusText

Reply:
this is easy to reproduce
You call the API with some missing required properties